tests/e2e/util.go

```diff
@@ -6823,3 +6826,90 @@ func getVmdkPathFromVolumeHandle(sshClientConfig *ssh.ClientConfig, masterIp str
+			if strings.Contains(servicesStoppedInVc[0], "StartPending:") {
+				pendingServiceslist = strings.Split(servicesStoppedInVc[0], "StartPending:")
+				pendingServiceslist = strings.Split(strings.Trim(pendingServiceslist[1], "\n "), " ")
+				framework.Logf("Additional service %s in StartPending state", pendingServiceslist)
+			}
+			servicesStoppedInVc = strings.Split(strings.Trim(servicesStoppedInVc[1], "\n "), " ")
+			// checking if current stopped services are same as defined above
+			if reflect.DeepEqual(servicesStoppedInVc, defaultStoppedServicesList) {
+				framework.Logf("All vc required services are in up and running state")
+				noAdditionalServiceStopped = true
+				// wait for 1 min,in case dependent service are still in pending state
+				time.Sleep(1 * time.Minute)
+			} else {
+				for _, service := range servicesStoppedInVc {
+					if !(slices.Contains(defaultStoppedServicesList, service)) {
+						framework.Logf("Starting additional service %s in stopped state", service)
+						sshCmd := fmt.Sprintf("service-control --start %s", service)
+						framework.Logf("Invoking command %v on vCenter host %v", sshCmd, host)
+						result, err := fssh.SSH(ctx, sshCmd, host, framework.TestContext.Provider)
+						if err != nil || result.Code != 0 {
+							fssh.LogResult(result)
+							return false, fmt.Errorf("couldn't execute command: %s on vCenter host: %v", sshCmd, err)
+						}
+						// wait for 30 seconds,in case dependent service are still in pending state
+						time.Sleep(30 * time.Second)
+					}
+				}
+			}
+			// Checking status for pendingStart Service list and starting it accordingly
+			for _, service := range pendingServiceslist {
+				framework.Logf("Checking status for additional service %s in StartPending state", service)
+				sshCmd := fmt.Sprintf("service-control --status %s", service)
+				framework.Logf("Invoking command %v on vCenter host %v", sshCmd, host)
+				result, err := fssh.SSH(ctx, sshCmd, host, framework.TestContext.Provider)
+				if err != nil || result.Code != 0 {
+					fssh.LogResult(result)
+					return false, fmt.Errorf("couldn't execute command: %s on vCenter host: %v", sshCmd, err)
+				}
+				if strings.Contains(strings.TrimSpace(result.Stdout), "Running") {
+					framework.Logf("Additional service %s got into Running from StartPending state", service)
+				} else {
+					framework.Logf("Starting additional service %s in StartPending state", service)
+					sshCmd = fmt.Sprintf("service-control --start %s", service)
+					framework.Logf("Invoking command %v on vCenter host %v", sshCmd, host)
+					result, err = fssh.SSH(ctx, sshCmd, host, framework.TestContext.Provider)
+					if err != nil || result.Code != 0 {
+						fssh.LogResult(result)
+						return false, fmt.Errorf("couldn't execute command: %s on vCenter host: %v", sshCmd, err)
+					}
+					// wait for 2 mins,in case dependent service are still in pending state
+					time.Sleep(30 * time.Second)
```

Comment:
your logic comment is not matching with static sleep time

Reply:
corrected